src/AWSMetadataUtilities.jl

```diff
@@ -567,34 +569,43 @@ function _generate_high_level_definition(
     - `String`: Docstring for the function
     """
     function _generate_docstring(function_name, documentation, required_parameters, optional_parameters)
+        function_name = _format_name(function_name)
+        args = join((_format_name(key) for (key, val) in required_parameters), ", ")
+        maybejoin = isempty(args) ? "" : ", "
         operation_definition = """
-            $(repeat('"', 3))
-                $function_name()
+            \"\"\"
+                $function_name($(args))
+                $function_name($(args)$(maybejoin)params::Dict{String,<:Any})
 
             $documentation\n
             """
 
         # Add in the required parameters if applicable
         if !isempty(required_parameters)
-            operation_definition *= "# Required Parameters\n"
+            operation_definition *= "# Arguments\n"
 
             for (required_key, required_value) in required_parameters
-                operation_definition *= "- `$required_key`: $(required_value["documentation"])\n"
+                key = _format_name(required_key)
+                operation_definition *= "- `$key`: $(required_value["documentation"])\n"
             end
 
             operation_definition *= "\n"
         end
 
         # Add in the optional parameters if applicable
         if !isempty(optional_parameters)
-            operation_definition *= "# Optional Parameters\n"
+            operation_definition *= """
+                # Optional Parameters
+                Optional parameters can be passed as a `params::Dict{String,<:Any}`. Valid keys are:
+                """
+
 
             for (optional_key, optional_value) in optional_parameters
-                operation_definition *= "- `$optional_key`: $(optional_value["documentation"])\n"
+                operation_definition *= "- `\"$optional_key\"`: $(optional_value["documentation"])\n"
             end
         end
 
-        operation_definition *= repeat('"', 3)
+        operation_definition *= "\"\"\""
```

Comment:
I would revert this back, I think it's a bit easier to read in the original form

```suggestion
        operation_definition *= repeat('"', 3)
```

Reply:
this confused both me and my editor 

but i can change it back if you feel strongly?

Reply:
Oh interesting, which editor are you using?
I'm not too too set on the `repeat()`, I find that reading quotes in strings to be a bit difficult at times to read, especially when modifying it.

Reply:
actually i think it wasn't my editor that was confused by it, but JuliaFormatter (which means probably CSTParser)

Reply:
yeah, on AWS.jl `master`, `using JuliaFormatter; format("src/AWSMetadataUtilities.jl")` errors (due to this `repeat` syntax afaict)

Reply:
I think that should be an issue logged w/ JuliaFormatter as it's valid?

Reply:
true